src/components/modals/save-map-modal.js

```diff
@@ -0,0 +1,144 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import {MAP_THUMBNAIL_DIMENSION} from 'constants/default-settings';
+
+import {
+  StyledModalContent,
+  InputLight,
+  TextAreaLight,
+  StyledModalVerticalPanel,
+  StyledModalSection,
+  StyledModalInputFootnote
+} from 'components/common/styled-components';
+import ImagePreview from 'components/common/image-preview';
+
+const StyledSaveMapModal = styled.div.attrs({
+  className: 'save-map-modal'
+})`
+  .save-map-modal-content {
+    min-height: 400px;
+  }
+
+  .modal-vertical-panel {
+    width: 40%;
+  }
+
+  .save-map-modal-description {
+    .modal-section-subtitle {
+      margin-left: 6px;
+    }
+  }
+`;
+
+const LIMIT = 100;
+
+function SaveMapModalFactory() {
+  class SaveMapModal extends Component {
+    static propTypes = {
+      exportImage: PropTypes.object.isRequired,
+      mapInfo: PropTypes.object.isRequired,
+      onSetMapInfo: PropTypes.func.isRequired,
+      thumbWidth: PropTypes.number.isRequired,
+      thumbHeight: PropTypes.number.isRequired
+    };
+
+    static defaultProps = {
+      thumbWidth: MAP_THUMBNAIL_DIMENSION.width,
+      thumbHeight: MAP_THUMBNAIL_DIMENSION.height,
+      destination: null
+    };
+
+    componentDidMount() {
+      this.props.onUpdateSetting({
+        mapW: this.props.thumbWidth,
+        mapH: this.props.thumbHeight
+      });
+    }
+
+    _onChangeInput = (key, limit, e) => {
+      const {
+        target: {value}
+      } = e;
+      // if (value.length <= limit) {
+      this.props.onSetMapInfo({[key]: value});
+      // }TextAreaLight
+    };
+
+    render() {
+      const {mapInfo, exportImage} = this.props;
+
+      return (
+        <StyledSaveMapModal>
+          <StyledModalContent className="save-map-modal-content">
+            <StyledModalVerticalPanel>
+              <StyledModalSection className="save-map-modal-name">
+                <div className="modal-section-title">Name*</div>
+                <div>
+                  <InputLight
+                    type="text"
+                    value={mapInfo.title}
+                    onChange={e => this._onChangeInput('title', LIMIT, e)}
```

Comment:
do we need to update map info for every single character, can't we simply pass all info when we submit?